database/influxdb/influxdb_retention_policy.py

```diff
@@ -0,0 +1,234 @@
+        for policy in retention_policies:
+            if policy['name'] == policy_name:
+                retention_policy = policy
+                break
+    except requests.exceptions.ConnectionError as e:
+        module.fail_json(msg=str(e))
+    return retention_policy
+
+
+def create_retention_policy(module, client):
+    database_name = module.params['database_name']
+    policy_name = module.params['policy_name']
+    duration = module.params['duration']
+    replication = module.params['replication']
+    default = module.params['default']
+
+    if not module.check_mode:
+        try:
+            client.create_retention_policy(policy_name, duration, replication, database_name, default)
+        except exceptions.InfluxDBClientError as e:
+            module.fail_json(msg=e.content)
+    module.exit_json(changed=True)
+
+
+def alter_retention_policy(module, client, retention_policy):
+    database_name = module.params['database_name']
+    policy_name = module.params['policy_name']
+    duration = module.params['duration']
+    replication = module.params['replication']
+    default = module.params['default']
+    duration_regexp = re.compile('(\d+)([hdw]{1})|(^INF$){1}')
+    changed = False
+
+    duration_lookup = duration_regexp.search(duration)
+
+    if duration_lookup.group(2) == 'h':
+        influxdb_duration_format = '%s0m0s' % duration
+    elif duration_lookup.group(2) == 'd':
+        influxdb_duration_format = '%sh0m0s' % (int(duration_lookup.group(1)) * 24)
+    elif duration_lookup.group(2) == 'w':
+        influxdb_duration_format = '%sh0m0s' % (int(duration_lookup.group(1)) * 24 * 7)
+    elif duration == 'INF':
+        influxdb_duration_format = 'INF'
+
+    if not module.check_mode:
+        if not retention_policy['duration'] == influxdb_duration_format or not retention_policy['replicaN'] == int(replication) or not retention_policy['default'] == default:
+            try:
+                client.alter_retention_policy(policy_name, database_name, duration, replication, default)
+                changed = True
+            except exceptions.InfluxDBClientError as e:
+                module.fail_json(msg=e.content)
+    module.exit_json(changed=changed)
+
+
+def main():
+    argument_spec = influxdb_argument_spec()
+    argument_spec.update(
+        policy_name=dict(required=True),
+        duration=dict(required=True),
+        replication=dict(required=True),
+        default=dict(default=False, type='bool')
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    if not HAS_INFLUXDB:
+        module.fail_json(msg='influxdb python package is required for this module')
+
+    client = connect_to_influxdb(module)
+    retention_policy = find_retention_policy(module, client)
+
+    if retention_policy:
+        alter_retention_policy(module, client, retention_policy)
```

Comment:
Best practice is to compare the contents of the retention policy and only take action (alter the retention policy) if a change is necessary. That change or lack thereof needs to be reported back to the controller.